bytecomp/translcore.ml

```diff
@@ -1221,7 +1221,10 @@ and transl_record env all_labels repres lbl_expr_list opt_init_expr =
         | Record_regular -> Lconst(Const_block(0, cl))
         | Record_inlined tag -> Lconst(Const_block(tag, cl))
         | Record_float ->
-            Lconst(Const_float_array(List.map extract_float cl))
+            if !Clflags.native_code then
```

Comment:
This change affects also the classic ("closure") mode.  Is this intentional?  Can this lead to any regression?